tests/cephfs/cephfs_utilsV1.py

```diff
@@ -816,6 +816,176 @@ def get_clone_status(self, client, vol_name, clone_name, **kwargs):
+        Args:
+            clients: Client_nodes
+            clone_obj:
+                "vol_name": Volume name where clone as been created Ex: cephfs,
+                "subvol_name": sub Volume name from where this clone has been created,
+                "snap_name": snapshot name,
+                "target_subvol_name": "clone_status_1",
+                "group_name": "subvolgroup_1",
+        """
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
+        clone_transistion_states = []
+        cmd_out, cmd_rc = self.get_clone_status(
+            client,
+            clone["vol_name"],
+            clone["target_subvol_name"],
+            group_name=clone.get("target_group_name", ""),
+        )
+        status = json.loads(cmd_out.read().decode())
+        if status["status"]["state"] not in clone_transistion_states:
+            clone_transistion_states.append(status["status"]["state"])
+        while status["status"]["state"] != expected_state:
+            cmd_out, cmd_rc = self.get_clone_status(
+                client,
+                clone["vol_name"],
+                clone["target_subvol_name"],
+                group_name=clone.get("target_group_name", ""),
+            )
+            status = json.loads(cmd_out.read().decode())
+            log.info(
+                f"Clone Status of {clone['vol_name']} : {status['status']['state']}"
+            )
+            if status["status"]["state"] not in [
+                "in-progress",
+                "complete",
+                "pending",
+                "canceled",
+            ]:
+                raise CommandFailed(f'{status["status"]["state"]} is not valid status')
+            if end_time < datetime.datetime.now():
+                raise CommandFailed(
+                    f"Clone creation has not reached to Complete state even after {timeout} sec"
+                    f'Current state of the clone is {status["status"]["state"]}'
+                )
+        return clone_transistion_states
+
+    def reboot_node(self, ceph_node, timeout=300):
+        ceph_node.exec_command(sudo=True, cmd="reboot", check_ec=False)
+        timeout = datetime.timedelta(seconds=timeout)
+        starttime = datetime.datetime.now()
+        while True:
+            try:
+                ceph_node.node.reconnect()
+                break
+            except BaseException:
+                if datetime.datetime.now() - starttime > timeout:
+                    log.error(
+                        "Failed to reconnect to the node {node} after reboot ".format(
+                            node=ceph_node.node.ip_address
+                        )
+                    )
+                    sleep(5)
+                    raise RuntimeError(
+                        "Failed to reconnect to the node {node} after reboot ".format(
+                            node=ceph_node.node.ip_address
+                        )
+                    )
```

Comment:
We could simplify this loop
```
endtime = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
while datetime.datetime.now() < endtime:
  try:
    node.reconnect()
    return
  except BaseException:
    sleep(5)

raise RuntimeError(...)
```

Reply:
Modified the function and ran the test case which uses it
Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-XXRXRA/